data_structures/linked_list.cpp

```diff
@@ -1,58 +1,147 @@
+ * node that the current node is pointing to, and then returning the current node to
+ * heap store.
+ */
 #include <iostream>
+#include <sstream>
 
-struct node {
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures {
+/**
+ * Class Documentation
+ */
+struct node
+{
     int val;
     node *next;
 };
+	
+/**
+ * This function checks if the string passed consists 
+ * of only digits.
+ *@param s To be checked if s contains only integers
+ *@returns true if there are only only digits present in the string
+ *@returns false if any other character is found
+ */
+bool isDigit(const std::string& s)
+{
+	//function statements here
+	for(int i=0;i<s.size()-1;i++)
+    {
+    	if(!isdigit(s[i]))
+    	{
+            return false;
+		}
+	}
+	return true;
+}
+	
+/**
+ * This function changes the value_type of its argument
+ * from a string to an integer
+ * @param s argument whose value_type we want to change
+ * @return number value with value_type as integer
+ */
+int toInt(const std::string& s)
+{
+	//function statements here
+	std::stringstream geek(s);
+	int number=0;
+	geek>>number;
+	return number;
+} 
 
-node *start;
+node *start=nullptr;
 
+/**
+ * This function creates a new node using the new value
+ * passed as its argument and inserts it to the end of the 
+ * list.
+ *@param x value we want to insert to the list
+ */
 void insert(int x) {
-    node *t = start;
-    node *n = new node;
-    n->val = x;
-    n->next = NULL;
-    if (start != NULL) {
-        while (t->next != NULL) {
+	//function statements here
+    node* t = start;
+    node* n = new node;//creates new node
+    n->val = x;//sets value(val) as x
+    n->next = nullptr;
+    if (t != nullptr) {
+        while (t->next != nullptr) {
             t = t->next;
-        }
-        t->next = n;
+        }//iterate to the end of the list
+        t->next = n;//current last node on the list is made
+	    //to point to the new node
     } else {
         start = n;
     }
 }
 
+/**
+ * This function removes an existing node from the current
+ * list containing the value passed as an argument to the 
+ * function.
+ * @param x value we want to remove from the list
+ */
 void remove(int x) {
-    if (start == NULL) {
+	//function statements here
+    if (start == nullptr) {
         std::cout << "\nLinked List is empty\n";
-        return;
-    } else if (start->val == x) {
-        node *temp = start;
+        return;//return if the list is empty 
+    } else if (start->val == x) {//check if the first node contains
+	    			//the value passed as an argument
         start = start->next;
-        delete temp;
         return;
     }
 
     node *temp = start, *parent = start;
-
-    while (temp != NULL && temp->val != x) {
+	//create iterators to the first and the next node
+	//and initialize both iterators with the first node
+	
+    //while the node pointed to by parent iterator 
```

Comment:
Same in other parts of the code (below and above).
```suggestion
    // while the node pointed to by parent iterator 
```